### PR TITLE
Return object format in expectation for yulOptimizerTests

### PR DESCRIPTION
**Pull request by axic**
Par of #9500.

### CODE REVIEW

File: test/libyul/YulOptimizerTest.cpp
Comment (axic) [inline test/libyul/YulOptimizerTest.cpp]: Could also rename every instance of `*m_ast` to `*m_object->code`, it is about 30 occurances.

File: test/libyul/yulOptimizerTests/controlFlowSimplifier/switch_remove_empty_case.yul
Comment (axic) [inline test/libyul/yulOptimizerTests/controlFlowSimplifier/switch_remove_empty_case.yul]: Did not update every test, but this is what it looks like. Majority (90%) of the tests do no have objects, but those which do result in a very weird uncompilable output.

We could also try to be smart and use code printing if there was only a code object and nothing.

**Comment by axic** [comment]
Need to squash this prior to merging.

File: test/libyul/YulOptimizerTest.cpp
Comment (axic) [inline test/libyul/YulOptimizerTest.cpp:359]: Here we are being smart.

**Review by chriseth — DISMISSED** [review]

**Comment by chriseth** [comment]
Please squash

**Comment by axic** [comment]
Merging as this was approved, only did a squash after approval as requested.

**Pull request merged**
